plugins/jobs/src/js/JobCreateEditFormModal.tsx

```diff
@@ -27,11 +45,14 @@ const JobCreateEditFormModal = componentFromStream<JobCreateEditFormModalProps>(
       errorMessage$.next(errorMessage);
     }
 
-    function handleSubmit(isEdit: boolean, jobSpec: JobData, jobId: string) {
-      const request$ = isEdit ? updateJob(jobId, jobSpec) : createJob(jobSpec);
+    function handleSubmit(
+      isEdit: boolean,
+      data: { jobId: string; jobSpec: JobData }
+    ) {
+      const mutation = isEdit ? updateJobMutation : createJobMutation;
 
-      combineLatest(props$, request$)
-        .take(1) // unsubscribe after the first one to break ∞ react update loop
+      combineLatest(props$, graphqlObservable(mutation, defaultSchema, data))
+        .first() // unsubscribe after the first one to break ∞ react update loop
```

Comment:
This comment is more like if you missed this discussion of first here: https://github.com/dcos/dcos-ui/pull/3028

(unfortunately, I lack the skill to get the link of the discussion :/)